src/app/hooks/usePermissions/permission_rules.ts

```diff
@@ -0,0 +1,130 @@
+import type {
+  Action,
+  AppointmentPermissionContext,
+  PermissionRule,
+  Role,
+  UserPermissionContext,
+} from "./types"
+
+/**
+ * Permission Sets - Common action combinations to reduce duplication
+ */
+const PERMISSION_SETS = {
+  READ_ONLY: ["view"] as Action[],
+  BASIC_CRUD: ["view", "create", "update", "delete"] as Action[],
+  MANAGEMENT: ["view", "create", "update", "approve", "assign", "cancel"] as Action[],
+  FULL_ADMIN: ["view", "create", "update", "delete", "approve", "assign", "cancel"] as Action[],
+  CREATE_READ: ["view", "create"] as Action[],
+  READ_UPDATE: ["view", "update"] as Action[],
+  USER_ACTIONS: ["view", "create", "update", "cancel"] as Action[],
+  STAFF_ACTIONS: ["view", "create", "update", "assign"] as Action[],
+  MANAGER_CRUD: ["view", "create", "update"] as Action[],
+} as const
+
+/**
+ * Condition Functions - Reusable permission conditions with proper typing
+ */
+const CONDITIONS = {
+  isOwnResource: (ctx: AppointmentPermissionContext | UserPermissionContext) => {
+    const ownerId = ctx.data?.userId
```

Comment:
This assumes the context uses `userId`. If it were `user_id` or `user` this wouldn't work. It may be easier if we let the conditions be defined per permission based on the userid and context.